lib/mutant/integration/minitest.rb

```diff
@@ -0,0 +1,112 @@
+require 'minitest'
+
+# monkey patch for minitest
+module Minitest
+  # prevent autorun from running tests when the vm closes
+  def self.autorun
+  end
+end
+
+module Mutant
+  class Integration
+    # Minitest integration
+    class Minitest < self
+      # represents the data associated with a test
+      class TestCase
+        include Concord.new(:class_name, :test_method)
+        include Adamantium
+
+        attr_reader :class_name
+
+        def run_passes?(reporter)
+          ::Minitest::Runnable.run_one_method(klass, test_method, reporter)
+          reporter.passed?
+        end
+
+        def method_name
+          test_method[5..-1]
+        end
+        memoize :method_name
+
+        private
+
+        def klass
+          Mutant.constant_lookup(class_name)
+        end
+        memoize :klass
+      end
+
+      register 'minitest'
+
+      def initialize
+        @output = StringIO.new
+        @reporter = make_reporter(@output)
+      end
+
+      def setup
+        $LOAD_PATH << 'test' unless $LOAD_PATH.include?('test')
+        $LOAD_PATH << 'lib' unless $LOAD_PATH.include?('lib')
+        Dir.glob('./test/**/*_test.rb').each(&method(:require))
+        Dir.glob('./test/**/test_*.rb').each(&method(:require))
+        self
+      end
+
+      def call(tests)
+        examples = tests.map(&all_tests_index.method(:fetch)).to_set
+        start = Time.now
+
+        examples.each { |test| break unless test.run_passes?(@reporter) }
```

Comment:
It's debateable, but I wonder if this would be simpler:

```ruby
break unless examples.all? { |test| test.run_passes?(@reporter) }
```

`Enumerable#all?` still short circuits if one of the tests returns false, but the break (imho) is easier to see outside of the block.